pkg/kopia/utils.go

```diff
@@ -194,6 +194,9 @@ func GetDataStoreGeneralMetadataCacheSize(opt map[string]int) int {
 
 // MarshalKopiaSnapshot encodes kopia SnapshotInfo struct into a string
 func MarshalKopiaSnapshot(snapInfo *SnapshotInfo) (string, error) {
+	if snapInfo.ID == "" {
```

Comment:
Do you need a nil check here?

Perhaps a better pattern would be to add a receiving method:

```
func (si*SnapshotInfo) Validate() error{
    if si == nil {
        return errors.New("Invalid Kopia snapshot: cannot be nil")
    }
    if si.ID == "" {
        return errors.New("Invalid Kopia snaoshot: ID cannot be empty")
    }
    return nil
}
```

Reply:
@tdmanv makes sense. I have made the changes